src/ai/backend/manager/api/filebrowser.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from typing import Any, Iterable, Mapping, Tuple
+
+import aiohttp
+import aiohttp_cors
+import sqlalchemy as sa
+import trafaret as t
+from aiohttp import web
+
+from ai.backend.common.logging import BraceStyleAdapter
+
+from ..exceptions import InvalidArgument
+from ..models import vfolders
+from .auth import auth_required
+from .context import RootContext
+from .manager import READ_ALLOWED, server_status_required
+from .types import CORSOptions, WebMiddleware
+from .utils import check_api_params
+
+log = BraceStyleAdapter(logging.getLogger(__name__))
+
+VFolderRow = Mapping[str, Any]
+
+
+async def get_vfid(root_ctx: RootContext, host: str, name: str) -> str:
+    async with root_ctx.db.begin_readonly() as conn:
+        query = (
+            sa.select([vfolders.c.id])
+            .select_from(vfolders)
+            .where(vfolders.c.host == host and vfolders.c.name == name)
+        )
+        folder_id = await conn.scalar(query)
+
+        return folder_id.hex
+
+
+@auth_required
+@server_status_required(READ_ALLOWED)
+@check_api_params(
+    t.Dict(
+        {
+            t.Key("host"): t.String,
+            t.Key("vfolders"): t.List(t.String),
+        },
+    ),
+)
+async def create_or_update_filebrowser(
+    request: web.Request,
+    params: Any,
+) -> web.Response:
+
+    root_ctx: RootContext = request.app["_root.context"]
+    vfolders = []
+    host = params["host"]
+    for vfolder_name in params["vfolders"]:
+        vfolders.append(
+            {
+                "name": vfolder_name,
+                "vfid": await get_vfid(root_ctx, host, vfolder_name),
+            },
+        )
+    proxy_name, _ = root_ctx.storage_manager.split_host(host)
+    try:
+        proxy_info = root_ctx.storage_manager._proxies[proxy_name]
+    except KeyError:
+        raise InvalidArgument("There is no such storage proxy", proxy_name)
+    headers = {}
+    headers["X-BackendAI-Storage-Auth-Token"] = proxy_info.secret
```

Comment:
Please refrain from using raw string itself rather than referencing `AUTH_TOKEN_HDR` constant under `ai.backend.manager.models.storage`.
```suggestion
    headers = {AUTH_TOKEN_HDR: proxy_info.secret}
```